cmd/parallel-manager.go

```diff
@@ -136,7 +150,21 @@ func (p *ParallelManager) monitorProgress() {
 }
 
 func (p *ParallelManager) queueTask(fn func() URLs) {
-	p.queueCh <- fn
+	p.doQueueTask(task{fn: fn})
+}
+
+func (p *ParallelManager) queueTaskWithBarrier(fn func() URLs) {
+	p.doQueueTask(task{fn: fn, barrier: true})
+}
+
+func (p *ParallelManager) doQueueTask(t task) {
```

Comment:
So this will block *before* the task is executed and wait until it is done before it returns?

It would seem that this function cannot be called concurrently. Please document.